gensim/models/word2vec.py

```diff
@@ -578,24 +578,26 @@ def scale_vocab(self, min_count=None, sample=None, dry_run=False, keep_raw_vocab
             self.min_count = min_count
             self.sample = sample
             self.vocab = {}
-        drop_unique, drop_total, retain_total, original_total = 0, 0, 0, 0
+        retain_total = drop_total = drop_unique = 0
         retain_words = []
         for word, v in iteritems(self.raw_vocab):
             if keep_vocab_item(word, v, min_count, trim_rule=trim_rule):
                 retain_words.append(word)
                 retain_total += v
-                original_total += v
                 if not dry_run:
                     self.vocab[word] = Vocab(count=v, index=len(self.index2word))
                     self.index2word.append(word)
             else:
                 drop_unique += 1
                 drop_total += v
-                original_total += v
-        logger.info("min_count=%d retains %i unique words (drops %i)",
-                    min_count, len(retain_words), drop_unique)
-        logger.info("min_count leaves %i word corpus (%i%% of original %i)",
-                    retain_total, retain_total * 100 / max(original_total, 1), original_total)
+        original_unique_total = len(retain_words) + drop_unique
+        retain_unique_pct = len(retain_words) * 100 / max(original_unique_total, 1)
+        logger.info("min_count=%d retains %i unique words (%i%% of original %i, drops %i)",
+                    min_count, len(retain_words), retain_unique_pct, original_unique_total, drop_unique)
+        original_total = retain_total + drop_total
+        retain_pct = retain_total * 100 / max(original_total, 1)
```

Comment:
The earlier `from __future__ import division` already ensures a float result (rather than int) when appropriate.